### PR TITLE
MINIFICPP-1024 - Fix CentOS and Fedora docker builds

### PR DESCRIPTION
**Pull request by bakaid**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: docker/centos/Dockerfile [hunk 1]
```diff
@@ -35,7 +35,8 @@ ENV MINIFI_BASE_DIR /opt/minifi
 RUN mkdir -p $MINIFI_BASE_DIR 
 USER $USER
 
-RUN yum -y install java-1.8.0-openjdk java-1.8.0-openjdk-devel gcc g++ sudo git which maven
+RUN yum -y install java-1.8.0-openjdk java-1.8.0-openjdk-devel gcc g++ sudo git which maven make
+RUN yum -y install openssl openssl-devel openssl-static
```
Comment (bakaid) [inline docker/centos/Dockerfile]: This is intended to be temporary until the third party compilations are fixed in MINIFICPP-1022

Comment (phrocker) [reply docker/centos/Dockerfile]: Let's wait on 1022, then

File: docker/ContainerBuild.sh [hunk 2]
```diff
@@ -34,6 +37,7 @@ echo "CMake Source Directory: $CMAKE_SOURCE_DIR"
 echo "MiNiFi Package: $MINIFI_SOURCE_CODE"
 
 # Copy the MiNiFi source tree to the Docker working directory before building
+rm -rf $CMAKE_SOURCE_DIR/docker/${DISTRO_NAME}/minificppsource/
```
Comment (phrocker) [inline docker/ContainerBuild.sh]: good catch!

File: docker/ContainerBuild.sh [hunk 3]
```diff
@@ -18,6 +18,9 @@
 
 #!/bin/bash
 
+# Fail on errors
+set -e
```
Comment (phrocker) [inline docker/ContainerBuild.sh]: Good catch

File: docker/centos/Dockerfile (hunk 1 above)
Comment (bakaid) [reply docker/centos/Dockerfile]: Agreed, removed the hotfix.

**Comment by phrocker** [comment]
I'll merge once travis finishes, but I suspect it'll be red on centos. I think that's acceptable, so I'll merge after just to make sure nothing else breaks.

**Comment by bakaid** [comment]
> I'll merge once travis finishes, but I suspect it'll be red on centos. I think that's acceptable, so I'll merge after just to make sure nothing else breaks.

Yep, it will also be red on Fedora, hopefully for not too long. Thanks!

**Pull request merged**